AssertionError in Iterative refinement

`AssertionError: R.refinable`

In lines 71-73 we do:
```julia
approx_error = min(norm(q - σ(ndir, S)), dot(ndir, q - p1))
refinable = (approx_error > TOL(N)) && !(norm(p1-q, 2) < TOL(N) || norm(q-p2, 2) < TOL(N))
ap = LocalApproximation{N}(p1, d1, p2, d2, q, refinable, approx_error)
```
In line 185 we check:
```julia
if Ω.approx_list[i].err <= ɛ
```
Now it can happen that `refinable == false`, although `Ω.approx_list[i].err`, which is `approx_error` above, is `<= ɛ`.
Instead, we should check:
```julia
approx = Ω.approx_list[i]
if !approx.refinable || approx.err <= ɛ
```

@mforets: Do you agree?

--- a/src/Approximations/iterative_refinement.jl
+++ b/src/Approximations/iterative_refinement.jl
@@ -47,7 +47,8 @@ PolygonalOverapproximation{N}(S::LazySet) where {N<:Real} =
     PolygonalOverapproximation(S::LazySet, LocalApproximation{N}[])
 
 """
-    new_approx(S::LazySet, p1::Vector{N}, d1::Vector{N}, p2::Vector{N}, d2::Vector{N}) where {N<:Real}
+    new_approx(S::LazySet, p1::Vector{N}, d1::Vector{N}, p2::Vector{N},
+               d2::Vector{N}) where {N<:Real}
 
 ### Input
 
@@ -61,7 +62,8 @@ PolygonalOverapproximation{N}(S::LazySet) where {N<:Real} =
 
 A local approximation of `S` in the given directions.
 """
-function new_approx(S::LazySet, p1::Vector{N}, d1::Vector{N}, p2::Vector{N}, d2::Vector{N}) where {N<:Real}
+function new_approx(S::LazySet, p1::Vector{N}, d1::Vector{N}, p2::Vector{N},
+                    d2::Vector{N}) where {N<:Real}
     if norm(p1-p2, 2) < TOL(N)
         # this approximation cannot be refined and we set q = p1 by convention
         ap = LocalApproximation{N}(p1, d1, p2, d2, p1, false, zero(N))
@@ -69,7 +71,8 @@ function new_approx(S::LazySet, p1::Vector{N}, d1::Vector{N}, p2::Vector{N}, d2:
         ndir = normalize([p2[2]-p1[2], p1[1]-p2[1]])
         q = intersection(Line(d1, dot(d1, p1)), Line(d2, dot(d2, p2)))
         approx_error = min(norm(q - σ(ndir, S)), dot(ndir, q - p1))
-        refinable = (approx_error > TOL(N)) && !(norm(p1-q, 2) < TOL(N) || norm(q-p2, 2) < TOL(N))
+        refinable = (approx_error > TOL(N)) &&
+                     !(norm(p1-q, 2) < TOL(N) || norm(q-p2, 2) < TOL(N))
         ap = LocalApproximation{N}(p1, d1, p2, d2, q, refinable, approx_error)
     end
     return ap
@@ -93,7 +96,8 @@ The list of local approximations in `Ω` of the set `Ω.S` is updated in-place a
 the new approximation is returned by this function.
 """
 function addapproximation!(Ω::PolygonalOverapproximation,
-    p1::Vector{N}, d1::Vector{N}, p2::Vector{N}, d2::Vector{N})::LocalApproximation{N} where {N<:Real}
+                           p1::Vector{N}, d1::Vector{N}, p2::Vector{N},
+                           d2::Vector{N})::LocalApproximation{N} where {N<:Real}
 
     approx = new_approx(Ω.S, p1, d1, p2, d2)
     push!(Ω.approx_list, approx)
@@ -182,7 +186,8 @@ function approximate(S::LazySet{N},
 
     i = 1
     while i <= length(Ω.approx_list)
-        if Ω.approx_list[i].err <= ɛ
+        approx = Ω.approx_list[i]
+        if !approx.refinable || approx.err <= ɛ
             # if this approximation doesn't need to be refined, consider the next
             # one in the queue (counter-clockwise order wrt d1)
             # if the approximation is not refinable => continue
@@ -194,14 +199,15 @@ function approximate(S::LazySet{N},
 
             Ω.approx_list[i] = la1
 
-            redundant = inext > length(Ω.approx_list) ? false : (norm(la2.p1-Ω.approx_list[inext].p1) < TOL(N)) && (norm(la2.q-Ω.approx_list[inext].q) < TOL(N))
+            redundant = inext > length(Ω.approx_list) ? false :
+                (norm(la2.p1-Ω.approx_list[inext].p1) < TOL(N)) &&
+                (norm(la2.q-Ω.approx_list[inext].q) < TOL(N))
             if redundant
                 # if it is redundant, keep the refined approximation
                 Ω.approx_list[inext] = la2
             else
                 insert!(Ω.approx_list, inext, la2)
             end
-
         end
     end
     return Ω

